extra/mariabackup/xtrabackup.cc

```diff
@@ -2250,6 +2268,26 @@ innodb_init_param(void)
 		srv_undo_dir = (char*) ".";
 	}
 
+	compile_time_assert(SRV_FORCE_IGNORE_CORRUPT == 1);
+	/*
+	 * This option can be read both from the command line, and the
+	 * defaults file. The assignment should account for both cases,
+	 * and for "--innobackupex". Since the command line argument is
+	 * parsed after the defaults file, it takes precedence.
+	 */
+	srv_force_recovery |= (ulong) xtrabackup_innodb_force_recovery;
```

Comment:
If using an additional global variable is necessary, then this must be a normal assignment, that is,
```c++
	srv_force_recovery = xtrabackup_innodb_force_recovery;
```
The cast looks redundant (both variables are of the same type), and the `|=` could lead to surprising results if we later allowed other values of this parameter than 0 or 1. Maybe you wanted to write the following:
```c++
	if (xtrabackup_innodb_force_recovery) {
		srv_force_recovery = xtrabackup_innodb_force_recovery;
	}
```

Reply:
You're right! Didn't think of cases other than `1`. Thanks.